shell/platform/linux/fl_text_input_plugin.cc

```diff
@@ -290,17 +292,23 @@ static void im_commit_cb(FlTextInputPlugin* self, const gchar* text) {
   FlTextInputPluginPrivate* priv = static_cast<FlTextInputPluginPrivate*>(
       fl_text_input_plugin_get_instance_private(self));
   std::string text_before_change = priv->text_model->GetText();
+  flutter::TextRange composing_before_change =
+      priv->text_model->composing_range();
   flutter::TextRange selection_before_change = priv->text_model->selection();
+  gboolean was_composing = priv->text_model->composing();
 
   priv->text_model->AddText(text);
   if (priv->text_model->composing()) {
     priv->text_model->CommitComposing();
   }
 
   if (priv->enable_delta_model) {
-    flutter::TextEditingDelta delta = flutter::TextEditingDelta(
-        text_before_change, selection_before_change, text);
-    update_editing_state_with_delta(self, &delta);
+    flutter::TextRange replace_range =
+        was_composing ? composing_before_change : selection_before_range;
```

Comment:
```suggestion
        was_composing ? composing_before_change : selection_before_change;
```

Reply:
Oops - sorry about that!

Reply:
Ooops not sure how I missed that! Thanks for patching!